cg/models/flow_cell/flow_cell.py

```diff
@@ -250,52 +191,102 @@ def get_sample_sheet(self) -> SampleSheet:
 
-    def is_hiseq_x_delivery_started(self) -> bool:
-        """Check if delivery of Hiseq X flow cell is started."""
-        LOG.info("Check if delivery of data from Hiseq X sequence instrument is ready")
-        return self.hiseq_x_delivery_started_path.exists()
+    def has_demultiplexing_started_locally(self) -> bool:
+        """Check if demultiplexing has started path exists on the cluster."""
+        return self.demultiplexing_started_path.exists()
 
-    def is_hiseq_x(self) -> bool:
-        """Check if flow cell is Hiseq X."""
-        LOG.debug("Check if flow cell is Hiseq X")
-        return self.hiseq_x_flow_cell.exists()
+    def has_demultiplexing_started_on_sequencer(self) -> bool:
+        """Check if demultiplexing has started on the NovaSeqX machine."""
+        latest_analysis: Path = get_latest_analysis_path(self.path)
+        if not latest_analysis:
+            return False
+        return Path(
+            latest_analysis, DemultiplexingDirsAndFiles.DATA, DemultiplexingDirsAndFiles.BCL_CONVERT
+        ).exists()
 
     def is_flow_cell_ready(self) -> bool:
         """Check if a flow cell is ready for demultiplexing.
 
         A flow cell is ready if the two files RTAComplete.txt and CopyComplete.txt exists in the
         flow cell directory.
         """
-        LOG.info("Check if flow cell is ready for demultiplexing")
+        LOG.debug("Check if flow cell is ready for demultiplexing")
         if not self.is_sequencing_done():
             LOG.info(f"Sequencing is not completed for flow cell {self.id}")
             return False
         LOG.debug(f"Sequence is done for flow cell {self.id}")
         if not self.is_copy_completed():
-            LOG.info(f"Copy of sequence data is not ready for flow cell {self.id}")
+            LOG.debug(f"Copy of sequence data is not ready for flow cell {self.id}")
             return False
         LOG.debug(f"All data has been transferred for flow cell {self.id}")
         LOG.info(f"Flow cell {self.id} is ready for demultiplexing")
         return True
 
-    def __str__(self):
-        return f"FlowCell(path={self.path},run_parameters_path={self.run_parameters_path})"
+    def is_sequencing_done(self) -> bool:
+        """Check if sequencing is done.
+        This is indicated by that the file RTAComplete.txt exists.
+        """
+        return self.rta_complete_path.exists()
+
+    def is_copy_completed(self) -> bool:
+        """Check if copy of flow cell is done.
+        This is indicated by that the file CopyComplete.txt exists.
+        """
+        return self.copy_complete_path.exists()
+
+
+class DemultiplexedFlowCellData(FlowCellDirectoryData):
+    """This class represents a demultiplexed flow cell in the demultiplexed runs folder."""
+
+    @property
+    def is_demultiplexing_complete(self) -> bool:
+        return Path(self.path, DemultiplexingDirsAndFiles.DEMUX_COMPLETE).exists()
+
+
+class CleanFlowCellData(FlowCellDirectoryData):
```

Comment:
Will these hold something otherwise we can skip them

Reply:
Yes, they will hold the information on used in the cleaning of flow cells. I will check to what extent. For now this is more  to see if we all agree on this change. Then I will implement it!